test/unit/query_context_cache_test.js

```diff
@@ -0,0 +1,220 @@
+const MAX_CAPACITY = 5;
+
+function QueryContextElement (id,timestamp,priority,context) {
+  this.id = id;
+  this.timestamp = timestamp;
+  this.priority = priority;
+  this.context = context;
+}
+
+function TestingQCC () {
+  this.qcc = null;
+ 
+  this.expectedIDs;
+  this.expectedReadTimestamp;
+  this.expectedPriority;
+  
+  this.initCache = function () {
+    this.qcc = new QueryContextCache(MAX_CAPACITY);
+  };
+  
+  this.initCacheWithData = function () {
+    this.initCacheWithDataWithContext(CONTEXT);
+  };
+  
+  this.initCacheWithDataWithContext = function (Context) {
+    this.qcc = new QueryContextCache(MAX_CAPACITY);
+    this.expectedIDs = [];
+    this.expectedReadTimestamp = [];
+    this.expectedPriority = [];
+    for (let i = 0; i < MAX_CAPACITY; i++) {
+      this.expectedIDs[i] = BASE_ID + i;
+      this.expectedReadTimestamp[i] = BASE_READ_TIMESTAMP + i;
+      this.expectedPriority[i] = BASE_PRIORITY + i;
+      this.qcc.merge(new QueryContextElement(this.expectedIDs[i], this.expectedReadTimestamp[i], this.expectedPriority[i], Context));
+    }
+  };
+    
+  this.initCacheWithDataInRandomOrder = function () {
+    this.qcc = new QueryContextCache(MAX_CAPACITY);
+    this.expectedIDs = [];
+    this.expectedReadTimestamp = [];
+    this.expectedPriority = [];
+    for (let i = 0; i < MAX_CAPACITY; i++) {
+      this.expectedIDs[i] = BASE_ID + i;
+      this.expectedReadTimestamp[i] = BASE_READ_TIMESTAMP + i;
+      this.expectedPriority[i] = BASE_PRIORITY + i;
+    }
+  
+    this.qcc.merge(new QueryContextElement(this.expectedIDs[3], this.expectedReadTimestamp[3], this.expectedPriority[3], CONTEXT));
+    this.qcc.merge(new QueryContextElement(this.expectedIDs[2], this.expectedReadTimestamp[2], this.expectedPriority[2], CONTEXT));
+    this.qcc.merge(new QueryContextElement(this.expectedIDs[4], this.expectedReadTimestamp[4], this.expectedPriority[4], CONTEXT));
+    this.qcc.merge(new QueryContextElement(this.expectedIDs[0], this.expectedReadTimestamp[0], this.expectedPriority[0], CONTEXT));
+    this.qcc.merge(new QueryContextElement(this.expectedIDs[1], this.expectedReadTimestamp[1], this.expectedPriority[1], CONTEXT));
+  };
+
+  this.assertCacheData = function () {
+    this.assertCacheDataWithContext(CONTEXT);
+  };
+  
+  this.assertCacheDataWithContext = function (Context) {
+    const size = this.qcc.getSize();
+    assert.strictEqual(size,MAX_CAPACITY);
+    const elements = Array.from(this.qcc.getElements());
+    for (let i = 0; i < size; i++) {
+      assert.strictEqual(this.expectedIDs[i], elements[i].id);
+      assert.strictEqual(this.expectedReadTimestamp[i], elements[i].timestamp);
+      assert.strictEqual(this.expectedPriority[i], elements[i].priority);
+      assert.strictEqual(Context, elements[i].context);
+    }
+  };
+}
+
+describe('Query Context Cache Test', function () {
+  const testingQcc = new TestingQCC();
+
+  /** Test for empty cache */
+  it('test - the cache is empty',function () {
```

Comment:
let's format this new file with `npm lint:fix -- pathToThisFile`

Reply:
I also ran the lint fix before I pushed it. Is this something wrong?

Reply:
I expected the whitespaces to be properly handled but linter didn't handled that - we will need to reconfigure it in the future